src/lints/struct_deprecated_added.ron

```diff
@@ -0,0 +1,60 @@
+SemverQuery(
+    id: "struct_deprecated_added",
+    human_readable_name: "struct #[deprecated] added",
+    description: "A struct has been newly marked with #[deprecated].",
+    required_update: Minor,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Struct {
```

Comment:
As is, this query is perfect and does what it sets out completely correctly.

There's an opportunity to make this same query apply not just for structs, but also for enums and (in the future, when we expose them in the query API) unions as well. To do that, instead of only selecting structs here, you could use `ImplOwner`, which is a common supertype of `struct/enum/union`, as the items that are allowed to have an inherent `impl` block.

Then we'd just need to make sure that the "current" and the "baseline" arms of the query are also talking about the same _kind_ of item in addition to having the same `importable_path` value.

You can see an example here: https://github.com/obi1kenobi/cargo-semver-checks/blob/main/src/lints/inherent_method_must_use_added.ron#L15-L18